test/test_helper.rb

```diff
@@ -131,66 +131,60 @@ def self.create(problem)
     Models::Vrp.create(coerce(problem))
   end
 
-  def self.matrix_required(vrp)
-    return unless ENV['TEST_DUMP_VRP'].to_s == 'true' && vrp.name
+  def self.matrices_required(vrps, filename)
+    dump_file = "test/fixtures/#{filename}.dump"
+    dumped_data = Marshal.load(Zlib.inflate(File.read(dump_file))) if File.file?(dump_file) # rubocop: disable Security/MarshalLoad
 
-    WebMock.enable_net_connect!
-    vrp.compute_matrix
-    WebMock.disable_net_connect!
-    File.write("test/fixtures/#{vrp.name}.dump", Zlib.deflate(Marshal.dump(vrp)))
-  end
+    write_in_dump = []
+    vrps.each{ |vrp|
+      next if vrp.matrices.any?
 
-  def self.multipe_matrices_required(vrps, filename)
-    return unless ENV['TEST_DUMP_VRP'].to_s == 'true'
+      if dumped_data.nil? && ENV['TEST_DUMP_VRP'] != 'true'
+        # WebMock.enable_net_connect!
+        vrp.compute_matrix
+        # WebMock.disable_net_connect!
+      else
+        OptimizerWrapper.router.stub(:matrix, lambda { |url, mode, dimensions, row, column, options|
+          puts 'stubed fct'
+          if ENV['TEST_DUMP_VRP'] == 'true'
+            matrices =
+              OptimizerWrapper.router.send(:__minitest_stub__matrix, url, mode, dimensions, row, column, options) # call original method
```

Comment:
```suggestion
        Routers::RouterWrapper.stub_any_instance(:matrix, lambda { |url, mode, dimensions, row, column, options|
```
What I understand here, is that you overload the current instance of the router loaded by the during all the test suite. So, it cannot be unloaded at the end of the stub. While Routers::RouterWrapper.stub_any_instance is not related to the context, so it can be removed once the current context is finished